Parser/asdl_c.py

```diff
@@ -1140,8 +1140,188 @@ def visitModule(self, mod):
     {NULL}
 };
 
+static PyObject *
+ast_repr_max_depth(AST_object *self, int depth);
+
+static PyObject *
+ast_repr_list(PyObject *list, int depth)
+{
+    struct ast_state *state = get_ast_state();
+    if (state == NULL) {
+        return NULL;
+    }
+
+    Py_ssize_t length = PySequence_Size(list);
+    if (length < 0) {
+        return NULL;
+    } else if (length == 0) {
+        return PyObject_Repr(list);
+    }
+
+    _PyUnicodeWriter writer;
+    bool is_list = PyList_Check(list);
+    PyObject *items[2];
+
+    items[0] = PySequence_GetItem(list, 0);
+    if (!items[0]) {
+        return NULL;
+    }
+    if (length > 1) {
+        items[1] = PySequence_GetItem(list, length - 1);
+        if (!items[1]) {
+            return NULL;
```

Comment:
This leaks a reference to `items[0]`.

If I'm not mistaken, this reference is also leaked unconditionally below; we never DECREF the two objects in `items`.